src/idl_gen_general.cpp

```diff
@@ -1130,6 +1132,71 @@ void GenStruct(StructDef &struct_def, std::string *code_ptr) {
       code += "); }\n";
     }
   }
+  if (struct_def.has_key) {
+    bool is_string = key_field->value.type.base_type == BASE_TYPE_STRING;
+    std::string key_name = lang_.language == IDLOptions::kCSharp ? MakeCamel(key_field->name) : key_field->name + "()";
```

Comment:
Use `auto` for variable declarations where possible.
Keep within 80 columns, see the Google C++ Style Guide.